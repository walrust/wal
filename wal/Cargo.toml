[package]
name = "wal"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
wal-component = {path = "../wal-component"}
wal-vdom = {path = "../wal-vdom"}

[dependencies.web-sys]
version = "0.3.64"
features = [
    "console", 
    "Window",
    "Document",
    "Node",
    "Element",
    "Text",
]