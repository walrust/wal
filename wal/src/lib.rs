pub mod virtual_dom;
pub mod component;
pub mod app;
pub mod utils;