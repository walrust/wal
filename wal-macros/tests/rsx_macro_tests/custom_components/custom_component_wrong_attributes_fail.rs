use wal::{
    component::{behavior::Behavior, Component},
    virtual_dom::VNode,
};
use wal_macros::rsx;

include!("../utils/custom_components/custom_component_props_i32.rs");

fn main() {
    rsx! { <CustomComponentPropsI32 attr = 5 /> };
    rsx! { <CustomComponentPropsI32 props = 1 attr = 5 /> };
    rsx! { <CustomComponentPropsI32 key = "key" attr = 5 /> };
    rsx! { <CustomComponentPropsI32 props = 1 attr = 5 key="key" /> };
    rsx! { <CustomComponentPropsI32 props = 0 props = 1 /> };
    rsx! { <CustomComponentPropsI32 props = 0 key = "key" props = 1 /> };
    rsx! { <CustomComponentPropsI32 key = "key1" key = "key2" /> };
    rsx! { <CustomComponentPropsI32 key = "key1" props = 0 key = "key2" /> };
    rsx! { <CustomComponentPropsI32 props /> };
    rsx! { <CustomComponentPropsI32 props = /> };
    rsx! { <CustomComponentPropsI32 props = key = "key" /> };
    rsx! { <CustomComponentPropsI32 key /> };
    rsx! { <CustomComponentPropsI32 key = /> };
    rsx! { <CustomComponentPropsI32 key = props = 1 /> };
}
