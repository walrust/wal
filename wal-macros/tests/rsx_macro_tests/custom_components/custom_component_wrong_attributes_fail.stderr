error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:10:37
   |
10 |     rsx! { <CustomComponentPropsI32 attr = 5 /> };
   |                                     ^^^^

error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:11:47
   |
11 |     rsx! { <CustomComponentPropsI32 props = 1 attr = 5 /> };
   |                                               ^^^^

error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:12:49
   |
12 |     rsx! { <CustomComponentPropsI32 key = "key" attr = 5 /> };
   |                                                 ^^^^

error: Unsupported attribute `attr`. Custom components supports only `props` and `key` attributes
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:13:47
   |
13 |     rsx! { <CustomComponentPropsI32 props = 1 attr = 5 key="key" /> };
   |                                               ^^^^

error: Duplicate attribute `props`
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:14:47
   |
14 |     rsx! { <CustomComponentPropsI32 props = 0 props = 1 /> };
   |                                               ^^^^^

error: Duplicate attribute `props`
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:15:59
   |
15 |     rsx! { <CustomComponentPropsI32 props = 0 key = "key" props = 1 /> };
   |                                                           ^^^^^

error: Duplicate attribute `key`
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:16:50
   |
16 |     rsx! { <CustomComponentPropsI32 key = "key1" key = "key2" /> };
   |                                                  ^^^

error: Duplicate attribute `key`
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:17:60
   |
17 |     rsx! { <CustomComponentPropsI32 key = "key1" props = 0 key = "key2" /> };
   |                                                            ^^^

error: expected `=`
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:18:43
   |
18 |     rsx! { <CustomComponentPropsI32 props /> };
   |                                           ^

error: Expected a literal, a struct literal expression or an expression block
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:19:45
   |
19 |     rsx! { <CustomComponentPropsI32 props = /> };
   |                                             ^

error: Expected a literal, a struct literal expression or an expression block
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:20:45
   |
20 |     rsx! { <CustomComponentPropsI32 props = key = "key" /> };
   |                                             ^^^

error: expected `=`
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:21:41
   |
21 |     rsx! { <CustomComponentPropsI32 key /> };
   |                                         ^

error: Expected a literal or an expression block
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:22:43
   |
22 |     rsx! { <CustomComponentPropsI32 key = /> };
   |                                           ^

error: Expected a literal or an expression block
  --> tests/rsx_macro_tests/custom_components/custom_component_wrong_attributes_fail.rs:23:43
   |
23 |     rsx! { <CustomComponentPropsI32 key = props = 1 /> };
   |                                           ^^^^^
